src-plugins/reformat/medResliceViewer.cpp

```diff
@@ -719,7 +719,40 @@ void medResliceViewer::generateOutput(vtkImageReslice* reslicer, QString destTyp
     compensateForRadiologicalView<DATA_TYPE>(outputImage);
     correctOutputTransform<DATA_TYPE>(outputImage, reslicer->GetResliceAxes());
 
+    // Final output image
     outputData->setData(outputImage);
+
+    // Update metadata: Columns, Rows, Size, SliceThickness, Orientation and Origin.
+    // XSpacing, YSpacing and ZSpacing are not filled in itkDCMTKDataImageReader.cpp
+    // However, SliceThickness == ZSpacing, and X and Y spacings can be read by the user on the view.
+    medUtilities::setDerivedMetaData(outputData, inputData, "reformatted");
+
```

Comment:
The calculations of metadata below should be generalised to avoid duplication (as is already the case in the associated crop PR https://github.com/Inria-Asclepios/medInria-public/pull/436/files#diff-c7a10cf573cc883fc4cf6bcdd5d4865fR492).
A function ```medUtilitiesITK:::updateMetadata``` could be created for this.

Reply:
It's done

Reply:
There's no change here

Reply:
Sorry, these lines are now in a dedicated function `medResliceViewer::setOutputMetaData`. The comment is still here because i did not change this specific line, just move it in a function

Reply:
@fcollot i added comments. Because of github errors from this morning, you may have missed it

Reply:
No what I meant is that any derived data could potentially need to update metadata such as columns, rows, slice thickness, origin, orientation etc. in case it has changed from the original data. To avoid duplication (which is already the case since https://github.com/Inria-Asclepios/medInria-public/pull/436 has much of the same code below), these lines should be refactored into a function that could be called on all medAbstractData of type ITK. This is why I suggested the ```medUtilitiesITK:::updateMetadata``` function.

Reply:
Hum, i'm not sure it's a good idea: really few toolboxes change as here the dimension size, slice tickness, orientation and origin. Crop for instance does not change orientation. Filtering toolboxes do not change the size, only the intensities, so there are a lot of toolboxes which do not need a dedicated function. We could create a toolbox which only change the echo or repetition time with several data, thus only the echo or repetition time would be necessary to be changed.

Reply:
The CPU cost of updating the metadata even if it hasn't changed is negligible, and I think it is preferable to having duplicate code which these two PRs are already introducing, and in your examples there will be even more duplication.

Reply:
I'm looking at doing that, but i think that's not really useful (even if i love to standardize code, and avoid duplication), because it would be applied on 2 toolboxes only in the entire application, and with non useful code for one of the two

Reply:
i don't really understand. Do you want to call medUtilitiesITK::updateMetadata from the plugin, or from medUtilities::setDerivedMetaData ?

Reply:
Because we need to use ImageType, and it's perfect if we use for instance in a plugin:

```c++
    medUtilities::setDerivedMetaData(outputData, inputData, "reformatted");
    medUtilitiesITK::updateMetadata<ImageType>(outputData);
```

But if you want to put updateMetadata in setDerivedMetaData we need to compute again ImageType, and we can't use DISPATCH_ON_3D_PIXEL_TYPE in this function.